extension/src/plots/model/collect.ts

```diff
@@ -104,21 +115,54 @@ const getCustomPlotData = (
   } as CustomPlotData
 }
 
+const fillColorScale = (
+  colorScale: ColorScale | undefined,
+  experiments: Experiment[]
+) => {
+  const completeColorScale = {
+    domain: [...(colorScale?.domain || [])],
+    range: [...(colorScale?.range || [])]
+  }
+
+  for (const experiment of experiments) {
+    const { id } = experiment
+    if (completeColorScale.domain.includes(id)) {
+      continue
+    }
+    completeColorScale.domain.push(id)
+    completeColorScale.range.push('#4c78a8' as Color)
+  }
+  return completeColorScale
+}
+
 export const collectCustomPlots = ({
+  colorScale,
   plotsOrderValues,
   experiments,
   height,
   nbItemsPerRow
 }: {
+  colorScale: ColorScale | undefined
   plotsOrderValues: CustomPlotsOrderValue[]
   experiments: Experiment[]
   height: number
   nbItemsPerRow: number
 }): CustomPlotData[] => {
   const plots = []
 
+  const completeColorScale = fillColorScale(colorScale, experiments)
```

Comment:
[F] If we want to respect (only show) selected revision then we just need to drop this function.

Reply:
No strong opinion but could we default to show only selected and have some sort of toggle to "show all"?

Reply:
That's a bigger change. Will come back to it after we make some changes for #4620.